lib/model/game_state.dart

```diff
@@ -50,6 +51,11 @@ class GameState with ChangeNotifier {
     randomiser.shuffle(deck);
   }
 
+  void viewDiscards() {
+    ViewDiscardPage(this);
```

Comment:
Ah, yes - so this is at least part of the problem. We're in the state code here, which is separate from the UI - it has no access to the widget tree and no way of rendering components. So, although you've constructed a widget here, it won't actually do anything or render anywhere.

The way the `ScryingPage` gets shown is by using logic in `PlayPage`:

```
if (state.scrying.isNotEmpty) {
    return ScryingPage(state);
  } else if (state.result != null) {
    return GameEndPage(state.result);
  } else { ... }
```

So, all we want to do here is set a new boolean field called `viewingDiscard`. Then update the `PlayPage` to:

```
if (state.scrying.isNotEmpty) {
    return ScryingPage(state);
  } else if (state.viewingDiscard) {
    return ViewDiscard(state);
  } else if (state.result != null) {
    return GameEndPage(state.result);
  } else { ... }
```

This will at least get the `ViewDiscard` widget to show at the appropriate time.